include/llbuild/CAS/CASDatabase.h

```diff
@@ -50,8 +51,18 @@ class CASDatabase {
     std::future<llvm::ErrorOr<std::unique_ptr<CASObject>>> = 0;
 
   /// Write the given object into the database.
-  virtual auto put(std::unique_ptr<CASObject> object) ->
+  virtual auto putObject(std::unique_ptr<CASObject> object) ->
     std::future<llvm::ErrorOr<DataID>> = 0;
+
+  /// Write an object into the database.
+  inline auto put(std::unique_ptr<CASObject> object)
+    -> std::future<llvm::ErrorOr<DataID>>
+  {
+    return putObject(std::move(object));
```

Comment:
Ah ha, I see here why the unique_ptr makes sense... for in memory.  🤔 Saves a copy going in, but we still have to copy it coming out.  Probably too early to say for certain, but seems like there is potential a lot of copies happening here.